.github/workflows/mobile-preview-builds-dev.yml

```diff
@@ -0,0 +1,33 @@
+name: Trigger mobile preview builds for iOS and Android on main branch
+on:
+  push:
+    branches:
+      - dev
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  deploy-eas-ios-dev:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/eas-deploy-ios
+        with:
+          EXPO_APPLE_APP_SPECIFIC_PASSWORD: ${{ secrets.EXPO_APPLE_APP_SPECIFIC_PASSWORD }}
+          APPLE_ID: ${{ secrets.APPLE_ID }}
+          ASC_APP_ID: ${{ secrets.ASC_APP_ID }}
+          EXPO_TOKEN: ${{ secrets.EXPO_TOKEN }}
+          LEATHER_BOT: ${{ secrets.LEATHER_BOT }}
+          IS_MAIN_BRANCH: true
```

Comment:
Wonder if it might be a better api to pass `TARGET_BRANCH`?

Reply:
A string, then it's down to inside the action to determine how it should respond to the branch